lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    Command.execute cmd, user_id
```

Comment:
The only thing that I thought of this morning was that I think the command should be executed in the player process not in the caller.

The player process is the players context and I would imagine just about all commands would need that in keeping with your example

```elixir
%{
  player: %Character{},
  room: %Room{},
  # ...
}
```
if we execute the command in the player process we know the character and room I think, and not have to look this up. I think that is more natural starting point for the execution of command.

A player process crashing should be ok as well - I sort of view it like this, first stage of failing in the caller would be parsing the command, then we handle that, the second stage would be passing the parsed command to the player process and if that fails something failed in the game and it couldn't run the command, which may be player related...